### DISCUSSION TITLE
Tutorials/3d_segmentation/spleen_segmentation_3d.ipynb  repruductive problem

### DISCUSSION THREAD

**Discussion by lanslotttTT** [Project-MONAI/MONAI#4277] [Q&A] [answered]
Thansk for your excellent great job! When I tried to run the spleen_segmentation_3d.ipynb on the colab I found that every time I got different output ,why this happen and how to solve it thanks in advance!
![2022-05-16 12-51-43 的屏幕截图](https://user-images.githubusercontent.com/30655387/168522076-16e6a134-d88b-449b-bc62-cf38c3ec388f.png)
![2022-05-16 12-52-07 的屏幕截图](https://user-images.githubusercontent.com/30655387/168522120-d575941d-b88f-4bdc-9e34-f9b04e9e558b.png)

And I run same code on my local machine my loss start at 0.6591
![2022-05-16 13-26-37 的屏幕截图](https://user-images.githubusercontent.com/30655387/168525323-e99daa5b-ed4b-4dfe-aaf0-8c47c1854b21.png)

Here is my config
 ```python
MONAI version: 0.8.1
Numpy version: 1.21.4
Pytorch version: 1.8.1+cu111
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 71ff399a3ea07aef667b23653620a290364095b1

Optional dependencies:
Pytorch Ignite version: 0.4.8
Nibabel version: 3.2.1
scikit-image version: 0.18.3
Pillow version: 8.4.0
Tensorboard version: 2.8.0
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.9.1+cu111
tqdm version: 4.62.3
lmdb version: 1.3.0
psutil version: 5.9.0
pandas version: 1.3.4
einops version: 0.4.0
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.


```

Here is the config on colab:
```python
MONAI version: 0.9.dev2220
Numpy version: 1.21.6
Pytorch version: 1.11.0+cu113
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 1e9b0084b1d32b4a72f6e309bacde1390526d38f
MONAI __file__: /usr/local/lib/python3.7/dist-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.8
Nibabel version: 3.0.2
scikit-image version: 0.18.3
Pillow version: 7.1.2
Tensorboard version: 2.8.0
gdown version: 4.4.0
TorchVision version: 0.12.0+cu113
tqdm version: 4.64.0
lmdb version: 0.99
psutil version: 5.4.8
pandas version: 1.3.5
einops version: NOT INSTALLED or UNKNOWN VERSION.
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: NOT INSTALLED or UNKNOWN VERSION.

```

**Comment by lanslotttTT** [comment, marked as answer]
I guess it's because of the notebook execution logic of the collab  that cause the computing graph will be saved all the time.
 But there still a problem in Colab ,when I run the cell below ,I got different input 
 
![2022-05-16 13-51-30 的屏幕截图](https://user-images.githubusercontent.com/30655387/168528137-71714dc1-918d-41c8-90cd-df6b09f550ca.png)
![2022-05-16 13-52-44 的屏幕截图](https://user-images.githubusercontent.com/30655387/168528144-d8cc6027-a1ec-454d-8915-bbd42b45e1ae.png)
![2022-05-16 13-54-40 的屏幕截图](https://user-images.githubusercontent.com/30655387/168528147-feef30ed-9735-4ae4-a0e2-28ecd28d435e.png)

**Reply by wyli** [reply]
mainly as it's commented on the pytorch doc page, https://pytorch.org/docs/stable/notes/randomness.html#reproducibility 

> Completely reproducible results are not guaranteed across PyTorch releases, individual commits, or different platforms.
